consumer-rest-core/src/main/scala/za/co/absa/spline/consumer/rest/controller/LineageOverviewController.scala

```diff
@@ -25,29 +25,32 @@ import za.co.absa.spline.consumer.service.repo.LineageRepository
 import scala.concurrent.Future
 
 @RestController
-class LineageController @Autowired()(val repo: LineageRepository) {
+@Api(tags = Array("lineage"))
+class LineageOverviewController @Autowired()(val repo: LineageRepository) {
 
   import scala.concurrent.ExecutionContext.Implicits._
 
-  @GetMapping(Array("/lineage"))
+  @GetMapping(Array("/lineage-overview"))
   @ApiOperation(
-    value = "GET /lineage",
-    notes = "Returns a lineage overview of executionEvent ID"
+    value = "Get execution event lineage overview",
+    notes =
+      """
+        Returns a lineage overview for a given execution event.
+        The graph consists of nodes of two types: data sources and executed jobs.
+        The lineage describes the historical data flow only, that is the data flow that has affected and
+        is observed by the given execution event (at the time of the event)
+      """
   )
   def lineage(
     @ApiParam("Execution event ID")
-    @RequestParam("executionEventId")
-    executionEventId: String,
+    @RequestParam("eventId")
```

Comment:
Because this `eventId` is required, it seems to me, more of a `path variable` than a `request param`

Reply:
Originally I thought the same, but then after reading up on REST best practices I understood that it's less the matter of the param being optional or not, but rather the matter of resource meaning and identity.
The common recommendation is to use path to describe the resource, and use query parameters for filtering and query on the resource.
E.g. `/users/bob`, `/event/123`, '/file/foo` are resources identified by their primary business keys, while `/issues?status=closed` or `/salary-report?industry=IT&year=2019`  are the queries on the resources. So I thought that requesting lineage overview for the given event ID is rather a query than an identity.

Reply:
Right, the required Vs optional is a rule of thumb to make the same distinction you describe. I'm not sure how the lineage overview for a specific event is understood as a filter. The lineage overview of a specific event seems like a concrete resource to me. In my understanding the event is the resource.

Reply:
The lineage (graph) is a resource. More like a report. You could request lineage by evenID, execID, datasource URI and many other params. So there could be many valid combinations of parameters uniquely identifying a certain piece of lineage. But there is no such thing as a lineage ID. Lineage is uncountable thing.

Reply:
I see! Then it makes sense. Thanks for clarifying :)